### PR TITLE
Item 10394: Updates for incorporating ELN in LKSM

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
The previous stories to separate the @labkey/eln package and other steps in the epic are complete and we are ready to wire up the ELN related features in LKSM. This will initially be done with an experimental feature. 

This PR adds a getPrimaryProductForContainer() helper for ProductRegistry. This can be used in the labbook module notifications to determine what URL to generate for the app links in the email templates.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/843
* https://github.com/LabKey/labbook/pull/186
* https://github.com/LabKey/biologics/pull/1329
* https://github.com/LabKey/sampleManagement/pull/967
* https://github.com/LabKey/platform/pull/3381
* https://github.com/LabKey/inventory/pull/439

#### Changes
* add getPrimaryProductForContainer() helper for ProductRegistry

### CODE REVIEW

File: core/api-src/org/labkey/api/products/ProductRegistry.java [hunk 1]
```diff
@@ -101,6 +102,13 @@ public List<String> getProductIdsForContainer(@NotNull Container container)
         return productIds;
     }
 
+    @Nullable
+    public ProductMenuProvider getPrimaryProductForContainer(@NotNull Container container)
```
Comment (labkey-nicka) [inline core/api-src/org/labkey/api/products/ProductRegistry.java]: nit: Consider renaming something like `getPrimaryProductMenuForContainer`.

File: core/api-src/org/labkey/api/products/ProductRegistry.java [hunk 2]
```diff
@@ -101,6 +102,13 @@ public List<String> getProductIdsForContainer(@NotNull Container container)
...
+    {
+        Set<String> modules = container.getActiveModules().stream().map(Module::getName).collect(Collectors.toSet());
```
Comment (labkey-nicka) [inline core/api-src/org/labkey/api/products/ProductRegistry.java]: Did you consider using the `productId`? Something like:

```java
@Nullable
public ProductMenuProvider getPrimaryProductForContainer(@NotNull Container container)
{
    List<String> productIds = getProductIdsForContainer(container);
    return getRegisteredProducts().stream().filter(product -> productIds.contains(product.getProductId())).findFirst().orElse(null);
}
```

**Review by labkey-nicka — APPROVED** [review]

File: core/api-src/org/labkey/api/products/ProductRegistry.java (hunk 1 above)
Comment (cnathe) [reply core/api-src/org/labkey/api/products/ProductRegistry.java]: done

File: core/api-src/org/labkey/api/products/ProductRegistry.java (hunk 2 above)
Comment (cnathe) [reply core/api-src/org/labkey/api/products/ProductRegistry.java]: I don't know why I was stuck on using the active modules for a container. This is much cleaner. Done.

**Pull request merged**